pkg/amphoracontrollers/lb_mgmt_network.go

```diff
@@ -0,0 +1,222 @@
+	}
+
+	var createOpts subnets.CreateOpts
+	if ipVersion == 6 {
+		gatewayIP := LbMgmtSubnetIPv6GatewayIP
+		createOpts = subnets.CreateOpts{
+			Name:            LbMgmtSubnetName,
+			Description:     LbMgmtSubnetDescription,
+			NetworkID:       lbMgmtNetID,
+			TenantID:        serviceTenantID,
+			CIDR:            LbMgmtSubnetIPv6CIDR,
+			IPVersion:       gophercloud.IPVersion(ipVersion),
+			IPv6AddressMode: LbMgmtSubnetIPv6AddressMode,
+			IPv6RAMode:      LbMgmtSubnetIPv6RAMode,
+			AllocationPools: []subnets.AllocationPool{
+				{
+					Start: LbMgmtSubnetIPv6AllocationPoolStart,
+					End:   LbMgmtSubnetIPv6AllocationPoolEnd,
+				},
+			},
+			GatewayIP: &gatewayIP,
+		}
+	} else {
+		gatewayIP := LbMgmtSubnetGatewayIP
+		createOpts = subnets.CreateOpts{
+			Name:        LbMgmtSubnetName,
+			Description: LbMgmtSubnetDescription,
+			NetworkID:   lbMgmtNetID,
+			TenantID:    serviceTenantID,
+			CIDR:        LbMgmtSubnetCIDR,
+			IPVersion:   gophercloud.IPVersion(ipVersion),
+			AllocationPools: []subnets.AllocationPool{
+				{
+					Start: LbMgmtSubnetAllocationPoolStart,
+					End:   LbMgmtSubnetAllocationPoolEnd,
+				},
+			},
+			GatewayIP: &gatewayIP,
+		}
+	}
+
+	var lbMgmtSubnet *subnets.Subnet
+	if len(allSubnets) == 0 {
+		log.Info(fmt.Sprintf("Creating Octavia management subnet \"%s\"", createOpts.Name))
+		lbMgmtSubnet, err = subnets.Create(client, createOpts).Extract()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		lbMgmtSubnet = &allSubnets[0]
+		delta := subnets.UpdateOpts{}
+		updateNeeded := false
+		if lbMgmtSubnet.Description != createOpts.Description {
+			delta.Description = &createOpts.Description
+			updateNeeded = true
+		}
+		if lbMgmtSubnet.AllocationPools[0].Start != createOpts.AllocationPools[0].Start ||
+			lbMgmtSubnet.AllocationPools[0].End != createOpts.AllocationPools[0].End {
+			delta.AllocationPools = []subnets.AllocationPool{
+				{
+					Start: createOpts.AllocationPools[0].Start,
+					End:   createOpts.AllocationPools[0].End,
+				},
+			}
+			updateNeeded = true
+		}
+		if lbMgmtSubnet.GatewayIP != *createOpts.GatewayIP {
+			delta.GatewayIP = createOpts.GatewayIP
+			updateNeeded = true
+		}
+		if updateNeeded {
+			log.Info(fmt.Sprintf("Updating Octavia management subnet \"%s\"", createOpts.Name))
+			lbMgmtSubnet, err = subnets.Update(client, lbMgmtSubnet.ID, delta).Extract()
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return lbMgmtSubnet, nil
+}
+
+func GetLbMgmtNetwork(client *gophercloud.ServiceClient, instance *octaviav1.OctaviaAmphoraController, serviceTenantID string) (*networks.Network, error) {
+	listOpts := networks.ListOpts{
+		Name:     LbMgmtNetName,
+		TenantID: serviceTenantID,
+	}
+	allPages, err := networks.List(client, listOpts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	allNetworks, err := networks.ExtractNetworks(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if len(allNetworks) > 0 {
+		return &allNetworks[0], nil
```

Comment:
This just returns the first network, not necessarily the lb-mgmt network, right?

Reply:
it's the first lb-mgmt network, `networks.List` is filtered by the name of the network (and also the tenant of the network)